src/components/date-range-picker/date-range-picker.ts

```diff
@@ -0,0 +1,1211 @@
+  }
+
+  // #endregion
+
+  // #region Private methods
+
+  protected _revertValue() {
+    this.value = this._oldValue;
+  }
+
+  private _getUpdatedDateRange(
+    input: IgcDateTimeInputComponent,
+    newValue: Date | null
+  ): DateRangeValue {
+    const currentStart = this.value?.start ?? null;
+    const currentEnd = this.value?.end ?? null;
+
+    return input === this._inputs[0]
+      ? { start: newValue, end: currentEnd }
+      : { start: currentStart, end: newValue };
+  }
+
+  // Delegates the validity methods of internal input elements
+  // to the component's own validation logic specific to date-range values.
+  // Checks for dirty state to avoid unnecessary validation on form reset,
+  // caused by the inputs value being set.
+  private _delegateInputsValidity() {
+    const inputs = this.useTwoInputs ? this._inputs : [this._input];
+
+    inputs.forEach((input) => {
+      input.checkValidity = () =>
+        this._dirty || !this._pristine ? this.checkValidity() : true;
+      input.reportValidity = () =>
+        this._dirty || !this._pristine ? this.reportValidity() : true;
+    });
+  }
+
+  private _setDateConstraints() {
+    const dates: DateRangeDescriptor[] = [];
+    if (this._min) {
+      dates.push({
+        type: DateRangeType.Before,
+        dateRange: [this._min],
+      });
+      if (this._inputs[0] && this._inputs[1]) {
+        this._inputs[0].min = this._min;
+        this._inputs[1].min = this._min;
+      }
+    }
+    if (this._max) {
+      dates.push({
+        type: DateRangeType.After,
+        dateRange: [this._max],
+      });
+      if (this._inputs[0] && this._inputs[1]) {
+        this._inputs[0].max = this._max;
+        this._inputs[1].max = this._max;
+      }
+    }
+    if (!isEmpty(this.disabledDates)) {
+      dates.push(...this.disabledDates);
+    }
+
+    this._dateConstraints = isEmpty(dates) ? [] : dates;
+  }
+
+  private _updateMaskedRangeValue() {
+    if (this.useTwoInputs) {
+      return;
+    }
+    if (!(this.value?.start && this.value?.end)) {
+      this._maskedRangeValue = '';
+      return;
+    }
+    let startMask = '';
+    let endMask = '';
+    const format =
+      DateTimeUtil.predefinedToDateDisplayFormat(this._displayFormat!) ??
+      this.displayFormat ??
+      this.inputFormat;
+    if (format) {
+      startMask = DateTimeUtil.formatDate(
+        this.value.start,
+        this.locale,
+        format
+      );
+      endMask = DateTimeUtil.formatDate(this.value.end, this.locale, format);
+    } else {
+      startMask = this.value.start.toLocaleDateString();
+      endMask = this.value.end.toLocaleDateString();
+    }
+    this._maskedRangeValue = `${startMask} - ${endMask}`;
+  }
+
+  private _setCalendarRangeValues() {
+    if (!this._calendar) {
+      return;
+    }
+
+    if (!(this.value?.start || this.value?.end)) {
+      this._calendar.values = null;
+      return;
+    }
+
+    if (!(this.value?.start && this.value?.end)) {
```

Comment:
I see a lot of checks for `value.start` and `value.end`. Can you abstract it in a helper getter/method?